qos/evm/request_validator.go

```diff
@@ -0,0 +1,170 @@
+package evm
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/pokt-network/poktroll/pkg/polylog"
+
+	"github.com/buildwithgrove/path/gateway"
+	qosobservations "github.com/buildwithgrove/path/observation/qos"
+	"github.com/buildwithgrove/path/qos/jsonrpc"
+)
+
+// maximum length of the error message stored in request validation failure observations and logs.
+const maxErrMessageLen = 1000
+
+// requestValidator handles EVM request validation, generating appropriate error contexts
+// when validation fails or request contexts when validation succeeds.
+type requestValidator struct {
+	logger        polylog.Logger
+	chainID       string
+	endpointStore *EndpointStore
+}
+
+// validateHTTPRequest validates an HTTP request, extracting and validating its JSONRPC payload.
+// If validation fails, an errorContext is returned along with false.
+// If validation succeeds, a fully initialized requestContext is returned along with true.
+func (rv *requestValidator) validateHTTPRequest(req *http.Request) (gateway.RequestQoSContext, bool) {
+	logger := rv.logger.With(
+		"qos", "EVM",
+		"method", "validateHTTPRequest",
+	)
+
+	// TODO_TECHDEBT(@adshmh): Simplify the qos package by refactoring gateway.QoSContextBuilder.
```

Comment:
Should this be `RequestQoSContext` or `QoSContextBuilder`?

Reply:
It should be `QoSContextBuilder`, which is the interface that currently receives an HTTP request.
The TODO asks for decoupling the builder from HTTP details, by defining a custom input struct.